src/schrodinger/sketcher/dialog/paste_in_text_dialog.h

```diff
@@ -36,15 +46,34 @@ class SKETCHER_API PasteInTextDialog : public ModalDialog
   protected:
     SketcherModel* m_sketcher_model = nullptr;
     std::unique_ptr<Ui::PasteInTextDialog> ui;
+    int m_autodetect_index = 0;
+    QString m_autodetect_base_name;
+    /**
+     * The autodetected format. Will be set to std::nullopt whenever the text
+     * edit is empty or when we can't determine the format.
+     */
+    std::optional<rdkit_extensions::Format> m_autodetected_format =
+        std::nullopt;
+    /**
+     * A map of all formats currently loaded into the Format combo box to their
+     * name
+     */
+    std::unordered_map<rdkit_extensions::Format, QString> m_formats;
```

Comment:
maybe a dumb question -- can you just use the combo box itself? why have a map as well?

Reply:
I could use the combo box itself, but that typically winds up being annoying and not very readable.  It's much easier to call ```m_formats.at(my_format)``` as opposed to something like ```ui->format_combo->itemText(ui->format_combo->findData(QVariant::fromValue(my_format)))```.  Also, if something goes wrong, it's much easier to debug a `std::out_of_range` exception from the `unordered_map::at` call than a mysteriously empty QString, which is what the later code would result in if `my_format` wasn't in the combo box.  I've been bitten by these types of issues enough when working with Python code that I tend to avoid directly interacting with combo box data unless I'm actively trying to affect the combo box itself.